src/request/components/request-detail-panel-execution.jsx

```diff
@@ -336,6 +338,60 @@ var ParamaterList = React.createClass({
     }
 });
 
+var MiddlewareList = React.createClass({
+    render: function() {
+        var middlewareStartMessages = this.props.middlewareStartMessages;
+        var middlewareEndMessages = this.props.middlewareEndMessages;
+        var beforeExecuteCommandMessages = this.props.beforeExecuteCommandMessages;
+        var afterExecuteCommandMessages = this.props.afterExecuteCommandMessages;
+        var mongoDBMessages = this.props.mongoDBMessages;
+        
+        var content = null;
+        
+        if (middlewareStartMessages
+            && middlewareEndMessages
+            && middlewareStartMessages.length >= 0
+            && middlewareEndMessages.length >= 0) {
+        
+            middlewareStartMessages = middlewareStartMessages.sort(function (a,b) { return a.ordinal - b.ordinal; });
+            middlewareEndMessages = _.indexBy(middlewareEndMessages, 'payload.correlationId');
+            
+            var middlewareComponent = _.map(middlewareStartMessages, function (middlewareStartMessage) {
+                var middlewareEndMessage = middlewareEndMessages[middlewareStartMessage.payload.correlationId];
+                var middlewareEndPayload = middlewareEndMessage.payload;
+                
+                var result = middlewareEndPayload.result;
+                
+                switch (result) {
+                    case 'next': result = 'Next'; break;
+                    case 'end': result = 'End'; break;
+                    case 'error': result = 'Error'; break;
+                }
```

Comment:
Why don't we just [title case](https://lodash.com/docs#startCase) this? It would work for the above cases as well as any other currently fall through cases.